### PR TITLE
add validating configs from configuration file

### LINKED ISSUE #21089: No validation of docker labels if using daemon configurations file
Hi, All,

Docker Daemon's labels need to be formatted as `key=value`. 
If we set labels in `docker daemon` with flags, docker will validate the format. 
While if labels are loaded from config file of `/etc/docker/daemon.json`, lables will escape from validating like below:
```
Labels:
 one
 two
```

Output of `docker version`:

```
root@ubuntu:/home/daocloud# docker version
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 21:37:01 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 21:37:01 2016
 OS/Arch:      linux/amd64
```

Output of `docker info`:

```
root@ubuntu:/home/daocloud# docker info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 38
Server Version: 1.10.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 44
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 3.19.0-25-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: ubuntu
ID: BSCV:H5FE:VVDN:KFVY:LIRY:NP6O:QR6K:OBGF:AL7Z:26Y3:HF5P:5C3R
WARNING: No swap limit support
Labels:
 one
 two
```

List the steps to reproduce the issue:
1.vi /etc/docker/daemon.json
```
{"labels": ["one", "two"]}
```
2.`service docker restart`
3.`docker info`

I think it is a bug that missed validating of labels.

### PR DESCRIPTION
**Pull request by allencloud**
1. Add config validation when these configs are from configuration file.
2. Add a test case in the test.File

fix #21089

### CODE REVIEW

**Comment by thaJeztah** [comment]
Think the failing test is related; https://jenkins.dockerproject.org/job/Docker-PRs/24958/console


```
10:01:44 ?   	github.com/docker/docker/contrib/httpserver	[no test files]
10:02:37 --- FAIL: TestValidateConfiguration (0.00s)
10:02:37 	config_test.go:243: valid DNS, got validation error
10:02:37 time="2016-03-11T10:02:37Z" level=warning msg="Changing requested CPUShares of 1 to minimum allowed of 2" 
10:02:37 time="2016-03-11T10:02:37Z" level=warning msg="Changing requested CPUShares of 262145 to maximum allowed of 262144" 
10:02:37 FAIL
10:02:37 coverage: 7.8% of statements
10:02:37 FAIL	github.com/docker/docker/daemon	0.074s
```

**Comment by thaJeztah** [comment]
ping @mlaventure @anusha-ragunathan the "experimental" and "gccgo" builds actually failed on this one, but were marked as "success"

https://jenkins.dockerproject.org/job/Docker-PRs-experimental/16148/console
https://jenkins.dockerproject.org/job/Docker-PRs-gccgo/3030/console

**Comment by allencloud** [comment]
@thaJeztah 
Thanks. I will check the test as soon as possible.

**Comment by anusha-ragunathan** [comment]
I cannot reproduce the test failure locally. It passes.

coverage: 0.0% of statements
ok      github.com/docker/docker/container      0.032s  coverage: 0.0% of statements
?       github.com/docker/docker/contrib/apparmor       [no test files]
?       github.com/docker/docker/contrib/docker-device-tool     [no test files]
?       github.com/docker/docker/contrib/httpserver     [no test files]
=== RUN   TestValidateConfiguration
--- PASS: TestValidateConfiguration (0.00s)
PASS

**Comment by anusha-ragunathan** [comment]
Also, the unit tests are configured the same (make.sh test-unit) across all jenkins jobs. Not sure why it would report as failed on one and passes on the others, when its failing on all jenkins jobs.@cpuguy83 did something change w.r.t status reporting in test infra?

File: daemon/config_test.go [hunk 1]
```diff
@@ -208,3 +208,71 @@ func TestFindConfigurationConflictsWithMergedValues(t *testing.T) {
...
+
+	c2 := &Config{
+		CommonConfig: CommonConfig{
+			Labels: []string{"one=two"},
+		},
+	}
+
+	err = validateConfiguration(c2)
+	if err != nil {
+		t.Fatal("valid Label, got validation error")
```
Comment (anusha-ragunathan) [inline daemon/config_test.go]: Useful to log err here.

**Comment by thaJeztah** [comment]
@anusha-ragunathan looking at the timeline, possibly @allencloud pushed an update before you tried to reproduce the failing test. Possibly the linked logs above can still give you a clue why those jobs were marked as "successful"

**Comment by anusha-ragunathan** [comment]
There was an issue in gccgo and experimental jenkins jobs. It's been fixed to report status correctly.

**Comment by thaJeztah** [comment]
@anusha-ragunathan thank you! Glad to hear you resolved that :clap:

File: daemon/config_test.go (hunk 1 above)
Comment (thaJeztah) [inline daemon/config_test.go]: @allencloud can you make that change?

Comment (thaJeztah) [inline daemon/config_test.go]: (perhaps for the other cases below as well)

Comment (allencloud) [inline daemon/config_test.go]: Thanks, @thaJeztah , I will change that as soon as possible.

**Comment by calavera** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by calavera** [comment]
:green_heart:

**Pull request merged**